controllers/imagepolicy_controller.go

```diff
@@ -284,3 +307,54 @@ func (r *ImagePolicyReconciler) patchStatus(ctx context.Context, req ctrl.Reques
 
 	return r.Status().Patch(ctx, &res, patch)
 }
+
+func (r *ImagePolicyReconciler) hasAccessToRepository(ctx context.Context, policy ctrl.Request, repo meta.NamespacedObjectReference, acl *imagev1.AccessFrom) (bool, error) {
+	// grant access if the policy is in the same namespace as the repository
+	if repo.Namespace == "" || policy.Namespace == repo.Namespace {
+		return true, nil
+	}
+
+	// deny access if the repository has no ACL defined
+	if acl == nil {
+		return false, fmt.Errorf("ImageRepository '%s/%s' can't be accessed due to missing access list",
+			repo.Namespace, repo.Name)
+	}
+
+	// grant access if the repository ACL has no namespace selectors
+	if acl != nil && acl.NamespaceSelectors == nil {
+		return true, nil
+	}
+
+	// get the policy namespace labels
+	var policyNamespace v1.Namespace
+	if err := r.Get(ctx, types.NamespacedName{Name: policy.Namespace}, &policyNamespace); err != nil {
+		return false, err
+	}
+	policyLabels := policyNamespace.GetLabels()
+
+	// deny access if the policy namespace has no labels
```

Comment:
This will deny access even if there's a selector that admits everything.

Reply:
Should be fixed by latest commit.